class-02-jquery-and-events/pair-assignment/class02navigator/scripts/articleView.js

```diff
@@ -0,0 +1,103 @@
+      //       and then show just the ones that match for the author that was selected.
+      //       Use an "attribute selector" to find those articles, and fade them in for the reader.
+      $('article').hide();
+      $('article[data-author="' + $(this).val() + '"]').fadeIn();
+
+    } else {
+      // DONE: If the select box was changed to an option that is blank, we should
+      //       show all the articles, except the one article we are using as a template.
+      $('article').fadeIn();
+      $('article.template').hide();
+    }
+    $('#category-filter').val('');
+  });
+};
+
+articleView.handleCategoryFilter = function() {
+  // DONE: Just like we do for #author-filter above, we should handle change events on the #category-filter element.
+  //       When an option with a value is selected, hide all the articles, then reveal the matches.
+  //       When the blank (default) option is selected, show all the articles, except for the template.
+  //       Be sure to reset the #author-filter while you are at it!
+  $('#category-filter').on('change', function() {
+    if ($(this).val()) {
+      $('article').hide();
+      $('article[data-category="' + $(this).val() + '"]').fadeIn();
+
+    } else {
+      $('article').fadeIn();
+      $('article.template').hide();
+    }
+    $('#category-filter').val('');
+  });
+};
+
+articleView.handleMainNav = function() {
+  // DONE: Add an event handler to .main-nav element that will power the Tabs feature.
+  //       Clicking any .tab element should hide all the .tab-content sections, and then reveal the
+  //       single .tab-content section that is associated with the clicked .tab element.
+  //       So: You need to dynamically build a selector string with the correct ID, based on the
+  //       data available to you on the .tab element that was clicked.
+  $('.main-nav li').on('click', function() {
+      $('.tab-content').hide();
+      $('.tab-content[data-content="' + $(this).text() + '"]');
+      $('#'+$(this).attr('data-content')).show();
+      console.log();
+  });
+
+  $('.main-nav .tab:first').click(); // Let's now trigger a click on the first .tab element, to set up the page.
+};
+
+articleView.setTeasers = function() {
+  $('.article-body *:nth-of-type(n+2)').hide(); // Hide elements beyond the first 2 in any artcile body.
+  // TODO: Add an event handler to reveal all the hidden elements,
+  //       when the .read-on link is clicked. You can go ahead and hide the
+  //       "Read On" link once it has been clicked. Be sure to prevent the default link-click action!
+  //       Ideally, we'd attach this as just 1 event handler on the #articles section, and let it
+  //       process any .read-on clicks that happen within child nodes.
+  $('.read-on').on('click', function (e){
+    e.preventDefault();
+      $(this).hide();
+      $('.article-body *:nth-of-type(n+2)').show();
```

Comment:
It shows the rest of the article but now that we went over it, we'll be better with selectors.